zio-query/shared/src/main/scala/zio/query/ZQuery.scala

```diff
@@ -657,6 +731,52 @@ object ZQuery {
       builder.map(_.result())
     }
 
+  /**
+   * Performs a query for each element in a collection, collecting the results
+   * into a query returning a collection of their results. Requests will be
+   * executed in parallel and will be batched.
+   *
+   * Queries will be split into `n` groups. Arbitrary effects within each group
+   * will not be performed in parallel but all requests to data source will
+   * still be batched.
+   */
+  def foreachParN[R, E, A, B, Collection[+Element] <: Iterable[Element]](n: Int)(
```

Comment:
The semantics of this are different than `foreachParN` on `ZIO`. Not sure if there is a better name for this.

Reply:
I think I prefer `*Batched` a little better than `*ParQuery`.

And this one could be `*BatchedParN`?

Also can you add the `collectAll` variant calling this one?

Reply:
Sounds good to me.